service/internal/checksum/middleware_setup_context.go

```diff
@@ -31,13 +40,17 @@ func (m *setupInputContext) HandleInitialize(
 ) (
 	out middleware.InitializeOutput, metadata middleware.Metadata, err error,
 ) {
-	// Check if validation algorithm is specified.
+	var algorithm string
+	var ok bool
 	if m.GetAlgorithm != nil {
```

Comment:
I know you're working around what was already there, but commenting since I noticed - is `m.GetAlgorithm == nil` actually possible? It seems like GetAlgorithm shouldn't ever be nil, if it were I think we'd have done something wrong in codegen.

I bring it up because the pre-declaration of <algorithm, ok> is a little awkward and wouldn't be necessary if this nil check wasn't there.

Reply:
checksum generator would generate getter for each op supporting checksum, I think we can remove that nil check